packagehandlers/mavenpackagehandler.go

```diff
@@ -153,15 +153,29 @@ type MavenPackageHandler struct {
 }
 
 func (mph *MavenPackageHandler) UpdateDependency(vulnDetails *utils.VulnerabilityDetails) error {
-	if err := mph.getProjectPoms(); err != nil {
+	depTreeOutput, clearMavenDepTreeRun, err := mph.runMavenDepTreeIfNeeded()
+	if err != nil {
 		return err
 	}
+
+	err = mph.getProjectPoms(depTreeOutput)
+	if err != nil {
+		return err
+	}
+
+	// If we got to this defer 'clearMavenDepTreeRun' contains a non-nil value
+	defer func() {
+		if clearMavenDepTreeRun != nil {
+			err = errors.Join(err, clearMavenDepTreeRun())
```

Comment:
for this line to work on defer you should change the signature to be (err error)